rhino/src/main/java/org/mozilla/javascript/ContextFactory.java

```diff
@@ -520,15 +519,4 @@ public final void exit() {
     public final Context enterContext(Context cx) {
         return Context.enter(cx, this);
     }
-
-    public void setTypeFactoryProvider(Function<Context, TypeInfoFactory> typeFactoryProvider) {
-        this.typeFactoryProvider = typeFactoryProvider;
-    }
-
-    public Function<Context, TypeInfoFactory> getTypeFactoryProvider() {
-        if (typeFactoryProvider == null) {
-            typeFactoryProvider = cx -> new ConcurrentFactory();
-        }
```

Comment:
Can you explain for what this is needed?
if you want to inject a existing TypeInfoFactory, you may associate to the scope before.